presto-main/src/test/java/com/facebook/presto/type/TestDate.java

```diff
@@ -193,4 +188,32 @@ public void testLeast()
         assertFunction("least(DATE '2013-03-30', DATE '2012-05-23')", DATE, new SqlDate(days));
         assertFunction("least(DATE '2013-03-30', DATE '2012-05-23', DATE '2012-06-01')", DATE, new SqlDate(days));
     }
+
+    @Test
+    public void dateCanBeImplicitlyCastedFromVarchar()
+            throws Exception
+    {
+        assertFunction("'1999-01-01' < DATE '2000-01-01'", BOOLEAN, true);
```

Comment:
Can you use `999` instead of `1999` - so lexegraphical order differs from chronological order?